entity-service/src/main/kotlin/com/egm/stellio/entity/authorization/Neo4jAuthorizationService.kt

```diff
@@ -28,26 +29,46 @@ class Neo4jAuthorizationService(
             .intersect(roles)
             .isNotEmpty()
 
-    override fun filterEntitiesUserCanRead(entitiesId: List<URI>, userSub: String): List<URI> =
-        filterEntitiesUserHaveOneOfGivenRights(entitiesId, READ_RIGHT, userSub)
+    override fun filterEntitiesUserCanRead(entitiesId: List<URI>, userSub: String): List<URI> {
+        val authorizedBySpecificPolicyEntities =
+            filterEntitiesWithSpecificAccessPolicy(
+                entitiesId,
+                listOf(SpecificAccessPolicy.AUTH_WRITE, SpecificAccessPolicy.AUTH_READ)
+            )
+        // remove the already authorized entities from the list to avoid double checking them
+        val grantedEntities = filterEntitiesUserHaveOneOfGivenRights(
+            entitiesId.minus(authorizedBySpecificPolicyEntities),
+            READ_RIGHT,
+            userSub
+        )
+        return authorizedBySpecificPolicyEntities.plus(grantedEntities)
+    }
 
-    override fun filterEntitiesUserCanUpdate(entitiesId: List<URI>, userSub: String): List<URI> =
-        filterEntitiesUserHaveOneOfGivenRights(entitiesId, WRITE_RIGHT, userSub)
+    override fun filterEntitiesUserCanUpdate(entitiesId: List<URI>, userSub: String): List<URI> {
+        val authorizedBySpecificPolicyEntities =
+            filterEntitiesWithSpecificAccessPolicy(entitiesId, listOf(SpecificAccessPolicy.AUTH_WRITE))
+        // remove the already authorized entities from the list to avoid double checking them
+        val grantedEntities = filterEntitiesUserHaveOneOfGivenRights(
+            entitiesId.minus(authorizedBySpecificPolicyEntities),
+            WRITE_RIGHT,
+            userSub
+        )
+        return authorizedBySpecificPolicyEntities.plus(grantedEntities)
+    }
 
     override fun filterEntitiesUserCanAdmin(entitiesId: List<URI>, userSub: String): List<URI> =
         filterEntitiesUserHaveOneOfGivenRights(entitiesId, ADMIN_RIGHT, userSub)
 
     override fun splitEntitiesByUserCanAdmin(
```

Comment:
Is it on purpose to keep arguments in separate lines ? I think one line is enough.

Reply:
where? I've not changed the line you're pointing to.

when I did it, it's only when it did not fit in the max line length.

Reply:
`splitEntitiesByUserCanAdmin` function, but indeed you did not changed the line.